src/easylink/rule.py

```diff
@@ -394,39 +410,43 @@ def build_rule(self) -> str:
         :class:`~CheckpointRule`. If this file does not yet exist when trying to
         aggregate, it means that the checkpoint has not yet been executed for the
         particular wildcard value(s). In this case, we manually raise a Snakemake
-        ``IncompleteCheckpointException`` which Snakemake automatically handles
+        ``IncompleteCheckpointException`` which `Snakemake automatically handles
+        <https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#data-dependent-conditional-execution>`_
         and leads to a re-evaluation after the checkpoint has successfully passed.
-
         """
-        aggregator = self._define_aggregator_function()
+        aggregator = self._define_input_function()
         rule = self._define_aggregator_rule()
         return aggregator + rule
 
-    def _define_aggregator_function(self):
-        """Builds the input data aggregator function."""
-        # FIXME: handle multiple filepaths
+    def _define_input_function(self):
+        """Builds the `input function <https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#input-functions>`_."""
+        if len(self.output_slot["filepaths"]) > 1:
```

Comment:
This feels like a slightly different thing; multiple files vs multiple slots?

Reply:
we only pass a single slot at a time into the AggregationRule.